src/network/networkd-route.c

```diff
@@ -777,3 +800,41 @@ int config_parse_route_scope(const char *unit,
 
         return 0;
 }
+
+int config_parse_route_table(const char *unit,
+                             const char *filename,
+                             unsigned line,
+                             const char *section,
+                             unsigned section_line,
+                             const char *lvalue,
+                             int ltype,
+                             const char *rvalue,
+                             void *data,
+                             void *userdata) {
+        _cleanup_route_free_ Route *n = NULL;
+        Network *network = userdata;
+        unsigned k;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = route_new_static(network, section_line, &n);
+        if (r < 0)
+                return r;
+
+        r = safe_atou(rvalue, &k);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, 0, "Routing table index out of range, must be in range 1..2^31, ignoring assignment: %s", rvalue);
```

Comment:
This message is wrong, because the error could be different:
```
r = safe_atou(rvalue, &k);
if (r < 0) {
    log_syntax(unit, LOG_ERR, filename, line, r,
                      "Could not parse route table number \"%s\", ignoring assignment: %m", rvalue);
    return 0;
}
if (k == 0 || k > INT_MAX) {
    log_syntax(unit, LOG_ERR, filename, line, 0,
                      "Routing table index out of range, must be in range 1..2^31, ignoring assignment: %s", rvalue);
    return 0;
}
```